includes/Core/REST_API/REST_Routes.php

```diff
@@ -423,7 +420,17 @@ function( $dataset ) use ( $module ) {
 									$responses = array_merge( $responses, $additional_responses );
 								}
 							}
-							return new WP_REST_Response( $this->parse_google_response_data( $responses ) );
+							$responses = $this->parse_google_response_data( $responses );
+							$responses = array_map(
+								function ( $response ) {
+									if ( is_wp_error( $response ) ) {
+										return $this->error_to_response( $response );
+									}
+									return $response;
+								},
+								$responses
+							);
```

Comment:
Shouldn't this happen before `parse_google_response_data()`? Since that might potentially JSON-encode and JSON-decode everything, messing up `WP_Error` objects.